libs/SalesforceSDK/src/com/salesforce/androidsdk/auth/NativeLoginManager.kt

```diff
@@ -266,6 +271,146 @@ internal class NativeLoginManager(
         return requestBodyString.toRequestBody(mediaType)
     }
 
+    // region Salesforce Identity API Headless Registration Flow
+
+    override suspend fun startRegistration(
+        email: String,
+        firstName: String,
+        lastName: String,
+        username: String,
+        newPassword: String,
+        reCaptchaToken: String,
+        otpVerificationMethod: OtpVerificationMethod
+    ): StartRegistrationResult {
+        // Validate parameters.
+        val trimmedEmail = email.trim()
+        if (!EMAIL_ADDRESS.matcher(trimmedEmail).matches()) {
```

Comment:
For email, it seemed easy enough to use Android's provided email pattern.  Any objections?